File: python/exercise10/self_checkout.py
items: list[dict[str, float | int]] = list()
price = 0.0
quantity = 0

for i in range(1, 4):
    price = float(input(f"Enter the price of item {i}: "))
    quantity = int(input(f"Enter the quantity of item {i}: "))

    items.append({"price": price, "quantity": quantity})


sub_total = 0.0
tax_rate = 5.5


for item in items:
    sub_total += item["price"] * item["quantity"]

tax = (tax_rate / 100) * sub_total
print(f"Subtotal: ${sub_total:.2f}")
print(f"Tax: ${tax:.2f}")
print(f"Total: ${(sub_total + tax):.2f}")

File: javascript/exercise10/self-checkout.js
import { input } from "pynput";

const items = new Map();
const price = 0.0;
const quantity = 0;

for (let i = 0; i < 3; i++) {
	const price = Number(await input(`Enter the price of item ${i + 1}: `));
	const quantity = Number(await input(`Enter the quantity of item ${i + 1}: `));

	items.set(i, { price, quantity });
}

let subtotal = 0.0;
const taxRate = 5.5;

for (const [_, value] of items) {
	subtotal += value.price * value.quantity;
}

const tax = (taxRate / 100) * subtotal;
console.log(`Subtotal: $${subtotal.toFixed(2)}`);
console.log(`Tax: $${tax.toFixed(2)}`);
console.log(`Total: $${(subtotal + tax).toFixed(2)}`);
